### PR TITLE
added payu latam for colombia gateway

### PR DESCRIPTION
**Pull request by damiangalindo**

### CODE REVIEW

**Comment by damiangalindo** [comment]
This is to add the functionality for PayU Latam for Colombia, but also gives the opportunity to add the other Latinamerican Gateways for PayU

File: lib/active_merchant/billing/gateways/payu_co.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module ActiveMerchant #:nodoc:
```
Comment (ntalbott) [inline lib/active_merchant/billing/gateways/payu_co.rb]: This doesn't seem to add value, since it's exactly the same as the values in PayuLatamGateway. Further, what is the difference besides currencies and countries between PayU Colombia and PayU Latam in general?

File: lib/active_merchant/billing/gateways/payu_latam.rb [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+        options.merge!({type: "REFUND"})
+
+        add_transaction(post, options)
+        
+        commit("refund", post)
+      end
+
+      # options = {order_id: "order_id", reason: "reson for the cancelation", transaction_id: "id of the transaction"}
+      def void(options={})
```
Comment (ntalbott) [inline lib/active_merchant/billing/gateways/payu_latam.rb]: `void` in ActiveMerchant is intended to reverse an `authorize`, so this doesn't seem useful without `authorize` support.

File: lib/active_merchant/billing/gateways/payu_latam.rb [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+            orderId: order_id
+          }
+        }
+
+        add_merchant(post)
+
+        commit("info", post)
+      end
+
+      def verify(credit_card, options={})
```
Comment (ntalbott) [inline lib/active_merchant/billing/gateways/payu_latam.rb]: I don't think this will work since there is no `authorize` support?

File: lib/active_merchant/billing/gateways/payu_latam.rb [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+        options.merge!({type: "VOID"})
+
+        add_transaction(post, options)
+        
+        commit("void", post)
+      end
+
+      #id for the order, integer value
+      def order_status(order_id)
```
Comment (ntalbott) [inline lib/active_merchant/billing/gateways/payu_latam.rb]: This is outside the scope of ActiveMerchant's public API. Please remove.

**Comment by ntalbott** [comment]
This is going to need some work before merging; see comments above. I'd love to see PayU Latam supported, though, so I'll try to stay on top of this!

**Comment by juankhazbun** [comment]
PayU Latam has support for Argentina, Colombia, México, Panamá, Perú and Brasil. Each of them have different currrency, so the idea of the PayuLatamGateway class is provide an easy way to extend the base functionality to the other supported countries.

Currently PayU Brasil is the only one that supports Authorize and Capture separately, as you can see in [Transaction Processing](http://developers.payulatam.com/en/sdk/payments.html) in PayU Latam page. Authorize and Capture together is the most used transaction type.  That's why we opted to extend a base PayuLatamGateway class so that we could reimplement any specific functionality based on the country.

**Comment by jasonwebster** [comment]
Closing due to staleness. Feel free to reopen when the outstanding issues are resolved.
